admin-dev/themes/new-theme/js/pages/product/components/combination-list/bulk-delete-handler.ts

```diff
@@ -69,43 +69,80 @@ export default class BulkDeleteHandler {
-      try {
-        const selectedCombinationsCount = this.bulkChoicesSelector.getSelectedCheckboxes().length;
-        const confirmLabel = bulkDeleteBtn.dataset.modalConfirmLabel
-          ?.replace(/%combinations_number%/, String(selectedCombinationsCount));
-
-        const modal = new ConfirmModal(
-          {
-            id: 'modal-confirm-delete-combinations',
-            confirmTitle: bulkDeleteBtn.innerHTML,
-            confirmMessage: bulkDeleteBtn.dataset.modalMessage,
-            confirmButtonLabel: confirmLabel,
-            closeButtonLabel: bulkDeleteBtn.dataset.modalCancelLabel,
-            closable: true,
-          },
-          async () => {
-            const response = await this.bulkDelete();
-            $.growl({message: response.message});
-            this.eventEmitter.emit(CombinationEvents.refreshCombinationList);
-          },
-        );
-        modal.show();
-      } catch (error) {
-        const errorMessage = error.response?.JSON ?? error;
-        $.growl.error({message: errorMessage});
-      }
+      this.bulkChoicesSelector.getSelectedIds().then((selectedCombinationIds) => {
+        try {
+          const selectedCombinationsCount = selectedCombinationIds.length;
+          const confirmLabel = bulkDeleteBtn.dataset.modalConfirmLabel
+            ?.replace(/%combinations_number%/, String(selectedCombinationsCount));
+
+          const modal = new ConfirmModal(
+            {
+              id: 'modal-confirm-delete-combinations',
+              confirmTitle: bulkDeleteBtn.innerHTML,
+              confirmMessage: bulkDeleteBtn.dataset.modalMessage,
+              confirmButtonLabel: confirmLabel,
+              closeButtonLabel: bulkDeleteBtn.dataset.modalCancelLabel,
+              closable: true,
+            },
+            async () => {
+              await this.bulkDelete(selectedCombinationIds);
+              //@todo: hardcoded success. Pass translated message to modal data attribute?
+              $.growl({message: 'Success'});
+              this.eventEmitter.emit(CombinationEvents.refreshCombinationList);
+            },
+          );
+          modal.show();
+        } catch (error) {
+          const errorMessage = error.response?.JSON ?? error;
+          $.growl.error({message: errorMessage});
+        }
+      });
     });
   }
 
   /**
    * @todo: the bulk delete action should be displayed with a progress modal once it is ready.
    */
-  private bulkDelete(): JQuery.jqXHR {
-    const combinationIds: number[] = [];
-    this.bulkChoicesSelector.getSelectedCheckboxes().forEach((checkbox: HTMLInputElement) => {
-      combinationIds.push(Number(checkbox.value));
-    });
+  private async bulkDelete(combinationIds: number[]): Promise<void> {
+    const progressModal = this.showProgressModal();
+    const progressModalElement = document.getElementById(CombinationMap.bulkProgressModalId);
+
+    let progress = 1;
+
+    //@todo: is there better loop option to avoid disabling eslint?
```

Comment:
Yeah I don't know why Eslint doesn't want us to use await in loops 

@NeOMakinG any idea why?

Reply:
Yes.

await inside a for loop will wait every promise to resolve one by one, it's time and performance consuming

You could do something like

```
const deletePromises = combinationIds.map(combinationId => this.combinationsService.deleteCombination(combinationId));

await Promise.all(deletePromises);
```

Reply:
Except in this case this is precisely what we aim to do, we want to execute the queries one at a time, wait for it to end, update the progress display, and then carry on with the next one.

The purpose is to perform a sequence of queries to avoid doing them all in parallel to avoid using too much server resource at once